CMakeLists.txt

```diff
@@ -5,15 +5,77 @@ project("SVF")
 configure_file(${PROJECT_SOURCE_DIR}/.config.in
                ${PROJECT_BINARY_DIR}/include/Util/config.h)
 
-# We need to match the build environment for LLVM: In particular, we need C++14
-# and the -fno-rtti flag
+# Match configuration used to build LLVM (match C++ standard; match
+# runtime typing information (RTTI); match exception handling; etc)
+if(COMMAND add_llvm_library)
+  message(STATUS "Detected in-tree build configuration; skipping LLVM fetching")
+  set(IN_SOURCE_BUILD 1)
+else()
+  message(STATUS "Detected out-of-tree build configuration; fetching LLVM")
+
+  find_package(LLVM REQUIRED CONFIG HINTS ${LLVM_DIR} $ENV{LLVM_DIR})
+  message(STATUS "LLVM STATUS:
+    Version       ${LLVM_PACKAGE_VERSION}
+    Definitions   ${LLVM_DEFINITIONS}
+    Includes      ${LLVM_INCLUDE_DIRS}
+    Libraries     ${LLVM_LIBRARY_DIRS}
+    Targets       ${LLVM_TARGETS_TO_BUILD}
+    Build type    ${LLVM_BUILD_TYPE}
+    Exceptions    ${LLVM_ENABLE_EH}
+    RTTI          ${LLVM_ENABLE_RTTI}
+    Dynamic lib   ${LLVM_LINK_LLVM_DYLIB}"
+  )
+
+  if("${LLVM_PACKAGE_VERSION_MAJOR}" VERSION_GREATER "15")
+    message(FATAL_ERROR "Unsupported LLVM version (need <= 15.0.7; got ${LLVM_PACKAGE_VERSION})")
+  endif()
+
+  if(CMAKE_BUILD_TYPE STREQUAL "Debug" AND NOT ${LLVM_BUILD_TYPE} STREQUAL "Debug")
+    message(FATAL_ERROR "Got build type 'Debug' but LLVM is not built in Debug mode!")
```

Comment:
We don't need LLVM to be built in debug mode if SVF is built under debug

Reply:
Ah, alright, fair enough. I added this check because of this remark [from this setup guide](https://github.com/SVF-tools/SVF/wiki/Setup-Guide-(Autoconf)):

> Build the debug version (make sure you have also installed the debug version of LLVM)

That page hasn't been updated since 2018 though, so I assume this is either no longer the case or was written erroneously and just never updated? Either way, I'll remove this check from the CMake script and update the setup guide page!